// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: model/oms_order_return_reason.proto

package mall

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on OrderReturnReason with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *OrderReturnReason) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on OrderReturnReason with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// OrderReturnReasonMultiError, or nil if none found.
func (m *OrderReturnReason) ValidateAll() error {
	return m.validate(true)
}

func (m *OrderReturnReason) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Id

	// no validation rules for Name

	// no validation rules for Sort

	// no validation rules for Status

	// no validation rules for CreateTime

	if len(errors) > 0 {
		return OrderReturnReasonMultiError(errors)
	}

	return nil
}

// OrderReturnReasonMultiError is an error wrapping multiple validation errors
// returned by OrderReturnReason.ValidateAll() if the designated constraints
// aren't met.
type OrderReturnReasonMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m OrderReturnReasonMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m OrderReturnReasonMultiError) AllErrors() []error { return m }

// OrderReturnReasonValidationError is the validation error returned by
// OrderReturnReason.Validate if the designated constraints aren't met.
type OrderReturnReasonValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e OrderReturnReasonValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e OrderReturnReasonValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e OrderReturnReasonValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e OrderReturnReasonValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e OrderReturnReasonValidationError) ErrorName() string {
	return "OrderReturnReasonValidationError"
}

// Error satisfies the builtin error interface
func (e OrderReturnReasonValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sOrderReturnReason.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = OrderReturnReasonValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = OrderReturnReasonValidationError{}
